pkg/server/options/options.go

```diff
@@ -114,15 +113,6 @@ func NewOptions(rootDir string) *Options {
 
 	// override all the stuff
 	o.GenericControlPlane.SecureServing.ServerCert.CertDirectory = rootDir
-	o.GenericControlPlane.Authentication = kubeoptions.NewBuiltInAuthenticationOptions().
-		WithAnonymous().
-		WithBootstrapToken().
-		WithClientCert().
-		WithOIDC().
-		WithRequestHeader().
-		WithServiceAccounts().
-		WithTokenFile().
-		WithWebHook()
```

Comment:
I don't get why this is removed. There's probably a reason, but I'm not sure what it is.

Reply:
Orginally, in #424, we started with 

```go
o.GenericControlPlane.Authentication = kubeoptions.NewBuiltInAuthenticationOptions().
		WithAnonymous().
		WithBootstrapToken().
		WithClientCert().
		WithOIDC().
		WithRequestHeader().
		// WithServiceAccounts().
		WithTokenFile()
	//WithWebHook()
```

In #681 the ServiceAccount option was re-enabled and in #3096 the webhook auth was re-enabled. So now the code in kcp is identical to upstream's `WithAll()` and so in order to cleanup and reduce cognitive load, I chose to remove the code here. I'm not sure if it was intended to not follow upstream.